lib/ansible/plugins/inventory/openshift.py

```diff
@@ -108,15 +108,86 @@
     context: 'awx/192-168-64-4:8443/developer'
 '''
 
-from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
-from ansible.module_utils.k8s.inventory import OpenShiftInventoryHelper
+from ansible.plugins.inventory.k8s import K8sInventoryException, InventoryModule as K8sInventoryModule, format_dynamic_api_exc
 
+try:
+    from openshift.dynamic.exceptions import DynamicApiError
+except ImportError:
+    pass
 
-class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable, OpenShiftInventoryHelper):
+
+class InventoryModule(K8sInventoryModule):
     NAME = 'openshift'
 
-    def parse(self, inventory, loader, path, cache=True):
-        super(InventoryModule, self).parse(inventory, loader, path)
-        cache_key = self._get_cache_prefix(path)
-        config_data = self._read_config_data(path)
-        self.setup(config_data, cache, cache_key)
+    helper = None
```

Comment:
what is this for?

Reply:
It comes from: https://github.com/ansible/ansible/blob/db5b65f5d618859ebb23c815797f7d88057c7fea/lib/ansible/module_utils/k8s/inventory.py#L246 Not used since 4d77878654e867c23d5f6c61422bdae7120393bc: I removed it.